pkg/certificates/config.go

```diff
@@ -0,0 +1,206 @@
+	}
+
+	identStatus := identConfig.Status()
+	if identStatus == identity.CertKey && !identConfig.Overwrite {
+		ident, err = identConfig.FullConfig().Load()
+		if err != nil {
+			return err
+		}
+	} else if identStatus != identity.NoCertNoKey && !identConfig.Overwrite {
+		return identity.ErrSetup.New("identity file(s) exist: %s", identStatus)
+	} else {
+		ident, err = identConfig.Create(ca)
+		if err != nil {
+			return err
+		}
+	}
+
+	signedChainBytes, err := c.Sign(ctx, ident)
+	if err != nil {
+		return errs.New("error occured while signing certificate: %s\n(identity files were still generated and saved, if you try again existnig files will be loaded)", err)
+	}
+
+	signedChain, err := identity.ParseCertChain(signedChainBytes)
+	if err != nil {
+		return nil
+	}
+
+	ca.Cert = signedChain[0]
+	ca.RestChain = signedChain[1:]
+	err = identity.FullCAConfig{
+		CertPath: caConfig.FullConfig().CertPath,
+	}.Save(ca)
+	if err != nil {
+		return err
+	}
+
+	ident.RestChain = signedChain[1:]
+	err = identity.Config{
+		CertPath: identConfig.FullConfig().CertPath,
+	}.Save(ident)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Sign submits a certificate signing request given the config
+func (c CertClientConfig) Sign(ctx context.Context, ident *identity.FullIdentity) ([][]byte, error) {
+	client, err := NewClient(ctx, ident, c.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.Sign(ctx, c.AuthToken)
+}
+
+// NewAuthDB creates or opens the authorization database specified by the config
+func (c CertServerConfig) NewAuthDB() (*AuthorizationDB, error) {
+	// TODO: refactor db selection logic?
+	driver, source, err := utils.SplitDBURL(c.AuthorizationDBURL)
+	if err != nil {
+		return nil, peertls.ErrRevocationDB.Wrap(err)
+	}
+
+	authDB := new(AuthorizationDB)
+	switch driver {
+	case "bolt":
+		_, err := os.Stat(source)
+		if c.Overwrite && err == nil {
+			if err := os.Remove(source); err != nil {
+				return nil, err
+			}
+		}
+
+		authDB.DB, err = boltdb.New(source, AuthorizationsBucket)
+		if err != nil {
+			return nil, ErrAuthorizationDB.Wrap(err)
+		}
+	case "redis":
+		redisClient, err := redis.NewClientFrom(c.AuthorizationDBURL)
+		if err != nil {
+			return nil, ErrAuthorizationDB.Wrap(err)
+		}
+
+		if c.Overwrite {
+			if err := redisClient.FlushDB(); err != nil {
+				return nil, err
+			}
+		}
+
+		authDB.DB = redisClient
+	default:
+		return nil, ErrAuthorizationDB.New("database scheme not supported: %s", driver)
+	}
+
+	return authDB, nil
+}
+
+// Run implements the responsibility interface, starting a certificate signing server.
+func (c CertServerConfig) Run(ctx context.Context, server *provider.Provider) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	authDB, err := c.NewAuthDB()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = errs.Combine(err, authDB.Close())
+	}()
+
+	signer, err := c.CA.Load()
+	if err != nil {
+		return err
+	}
+
+	srv := &CertificateSigner{
+		Log:           zap.L(),
```

Comment:
does `Log` need to be public? In most other stuff I've seen it declared as lowercase.